lib/config/configuration.go

```diff
@@ -618,13 +618,16 @@ func applyAuthConfig(fc *FileConfig, cfg *service.Config) error {
 		return trace.Wrap(err)
 	}
 
-	// Set session recording configuration from file configuration.
-	cfg.Auth.SessionRecordingConfig, err = types.NewSessionRecordingConfigFromConfigFile(types.SessionRecordingConfigSpecV2{
-		Mode:                fc.Auth.SessionRecording,
-		ProxyChecksHostKeys: fc.Auth.ProxyChecksHostKeys,
-	})
-	if err != nil {
-		return trace.Wrap(err)
+	// Only override session recording configuration if either field is
+	// specified in file configuration.
+	if fc.Auth.SessionRecording != "" || fc.Auth.ProxyChecksHostKeys != nil {
```

Comment:
If any of these is unset, should it still be included in the session recording config below?

Reply:
I was also curious about that because that's the existing behavior. We must have a `CheckAndSetDefaults` somewhere later because if it's unset, the default value gets populated.

Take a look at the test I added, that's actually what I was curious about too.